gossip/src/crds.rs

```diff
@@ -302,7 +307,8 @@ impl Crds {
                 Ok(())
             }
             Entry::Occupied(mut entry) => {
-                self.stats.lock().unwrap().record_fail(&value, route);
+                let mut stats = self.stats.lock().unwrap();
```

Comment:
We need to lock `self.stats` on every enum branch:
https://github.com/anza-xyz/agave/blob/80d3200f4/gossip/src/crds.rs#L240
https://github.com/anza-xyz/agave/blob/80d3200f4/gossip/src/crds.rs#L266
https://github.com/anza-xyz/agave/blob/80d3200f4/gossip/src/crds.rs#L305

So maybe move this to right before the `match` line:
https://github.com/anza-xyz/agave/blob/80d3200f4/gossip/src/crds.rs#L238
and remove it from the branches.